### PR TITLE
Move unsigned int operations to system module

### PR DESCRIPTION
**Pull request by def-**
This should lead to less confusion because uint numbers and literals seem to work, but operators are just missing.

Not sure if this is wanted, this just seems like a common gotcha.

### CODE REVIEW

**Comment by reactormonk** [comment]
I assume you can't move the unsigned literals to the unsigned module?

**Comment by dom96** [comment]
In my opinion this should be merged, but it's up to @Araq.

**Comment by oderwat** [comment]
It was the first thing which was like "?" for me as I tried Nim and converted some code. Then I read about the reasons and thought it's OK for me to "opt in" if I really need unsigned calculations. I would not mind if that's not needed though.

**Comment by Araq** [comment]
Though decision.

**Comment by def-** [comment]
My main motiviation is that newcomers have fewer surprises when coming to Nim. Maybe we should instead explain in the Tutorial why signed integers are the default and should be preferred?

**Comment by EXetoC** [comment]
@def- Yes that, and/or in a guidelines document. I think the current approach has made more people aware of how rarely unsigned types really are, and so I'm optimistic that it will be discussed in code reviews in cases where unsigned integers are used where signed integers would have sufficed.

If we do keep it necessary to request support for unsigned integers, then perhaps that should include everything, including types and literals, but it would be an intrusive change at this point.

**Comment by trustable-code** [comment]
Why should users be discouraged from using unsigned integers? (sorry for noob question)
Found this in the chat: Unsigned ints are guaranteed to wrap around, signed ints can do whatever they want in C.

**Comment by def-** [comment]
See for example:

- https://stackoverflow.com/questions/18795453/why-prefer-signed-over-unsigned-in-c
- http://www.eschertech.com/articles/items/art100407.html
- https://news.ycombinator.com/item?id=7693988

**Comment by reactormonk** [comment]
If possible we should move the unsigned literals to `unsigned` instead, so any usage of unsigned numbers is deliberate. Is that possible?

**Comment by Araq** [comment]
@reactormonk too much work to bother and breaks code.

**Comment by Araq** [comment]
Made up my mind, can you rebase please so that I can pull it?

**Comment by def-** [comment]
@Araq Done

File: lib/core/unsigned.nim [hunk 1]
```diff
@@ -7,51 +7,12 @@
 #    distribution, for details about the copyright.
 #
 
-## This module implements basic arithmetic operators for unsigned integers.
-## To discourage users from using ``unsigned``, it's not part of ``system``,
-## but an extra import.
+## **Warning:** Since version 0.10.4 this module is deprecated.
```
Comment (dom96) [inline lib/core/unsigned.nim]: Change to 0.11.4.

**Comment by Araq** [comment]
I think you need to update system/mmdisp too for the Go GC support.

**Comment by def-** [comment]
@Araq No idea what needs to be updated there.

**Pull request merged**